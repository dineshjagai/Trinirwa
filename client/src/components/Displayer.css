.wrapper {
    background: white;
    border-radius: 15px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.section2 {
    text-align: center;
}