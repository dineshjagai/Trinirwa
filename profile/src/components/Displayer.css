.section1 {
    display: flex;
    flex-wrap: wrap;
    flex-direction:vertical;
    justify-content: flex-start;
   

}

.wrapper {
   
    background: white;
    border-radius: 15px;
    width:25%;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

}

.section2 {
    text-align: center;
}