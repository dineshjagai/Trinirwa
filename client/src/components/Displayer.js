import React from 'react';
import './Displayer.css';
import 'bootstrap/dist/css/bootstrap.css';
import Divider from '@material-ui/core/Divider';
import DialogBox from './Dialog';

export default function Displayer({ interests, addInterest, deleteInterest }) {
  const items = [];
  interests.forEach((interest) => {
    items.push(
      <li key={interest} onClick={() => deleteInterest(interest)} className="badge rounded-pill bg-secondary m-2">
        {interest}
      </li>,
    );
  });

  return (
    <div className="wrapper">
      <div className="section1">
        <ul>{items}</ul>
      </div>
      <Divider variant="middle" />
      <div className="section2">
        <DialogBox onClose={addInterest} />
      </div>
    </div>

  );
}
