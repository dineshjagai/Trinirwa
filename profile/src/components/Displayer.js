import React, { useState} from 'react';
import ReactDOM from 'react-dom';
import './Displayer.css';
import 'bootstrap/dist/css/bootstrap.css';
import axios from "axios";
import Divider from '@material-ui/core/Divider';
import DialogBox from './Dialog.js'

export default function Displayer(props){
    const[interests, setInterests]= useState(props.interests);
    function handleDelete(interes){
        const delUrl= '/profile/delete/interest/'+ props.id;
        axios({
            method: 'DELETE',
            url: delUrl,
            data: {
                interest: interes,
            }
        });
        const Newinterests = interests.filter(interest=> interest != interes);
        setInterests(Newinterests);
    }

    const items = interests.map((interest)=>
        <li  onClick= {()=>handleDelete(interest)} className= "badge rounded-pill bg-secondary m-2" key={interest}>
            {interest}
        </li>
    );

    return(
        <div className="wrapper">
            <div className="section1">
                <ul >{items}</ul>
            </div>
            <Divider  variant="middle"/>
            <div className="section2">
                <DialogBox  id={props.id}/>
            </div>
        </div>
        
    );
}


