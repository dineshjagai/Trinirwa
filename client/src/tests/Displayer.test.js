import React from 'react';
import renderer from 'react-test-renderer';
import Displayer from '../components/Displayer';

test('Test Displayer snapshot', () => {
  const interest = ['me', 'you'];

  const functOne = () => {
    console.log("Add interests!");
  }

  const functTwo = () => {
    console.log("Delete interests!");
  }


  const component = renderer.create(<Displayer interests={interest} 
  addInterest={functOne} deleteInterest={functTwo} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
