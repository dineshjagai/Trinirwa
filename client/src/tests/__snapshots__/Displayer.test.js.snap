// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Displayer snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="section1"
  >
    <ul>
      <li
        className="badge rounded-pill bg-secondary m-2"
        onClick={[Function]}
      >
        me
      </li>
      <li
        className="badge rounded-pill bg-secondary m-2"
        onClick={[Function]}
      >
        you
      </li>
    </ul>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle"
  />
  <div
    className="section2"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Add interest
        </span>
      </button>
    </div>
  </div>
</div>
`;
